spaceR/spaceR/Controllers/LoginViewController.swift

```diff
@@ -27,18 +34,15 @@ class LoginViewController: UIViewController, UITextFieldDelegate {
     }
     
     @IBAction private func loginButtonClicked(_ sender: Any) {
-        if let safeEmail = self.textFieldEmailID.text, !safeEmail.isEmpty,
-           let safePassword = self.textFieldPassword.text, !safePassword.isEmpty {
-            if loginViewModel.validateUserCredentials(userEmail: safeEmail, userPassword: safePassword) {
-                successfulLogin()
-            } else {
-                unsuccessfulLogin()
-            }
-        } else {
+        do {
+            try loginViewModel.validateUserCredentials(userEmail: textFieldEmailID.text,
```

Comment:
Instead of having `loginViewModel.validateUserCredentials` return a `bool` why not let your `ViewModel` do the evaluation and then call on of your delegate functions based on the `ViewModels` evaluation.